File: Aula 5/principal.py
from clima import Clima

lista = []
nome_base = "base.csv"

mapa_preciptacao = {
    "nada": 0,
    "pouca": 1,
    "média": 2,
    "muita": 3
}
    

try: #tentar
    #abrindo o arquivo no modo leitura
    leitor = open(nome_base, "r", encoding="utf-8")

    for linha in leitor:
        ano, mes, temperatura, preciptacao = linha.strip().split(",") 

        obj_clima = Clima(ano, mes, temperatura, preciptacao) 

        if obj_clima not in lista:
            lista.append(obj_clima)
    
    leitor.close()
    
    for item in lista:
        print(item)

    maior = lista[0]

    for item in lista:
        if mapa_preciptacao[item.preciptacao] > mapa_preciptacao[maior.preciptacao]:
            maior = item

    print("Mes com maior preciptação: ")
    print(maior)

except Exception as e:
    print("Ocorreu um erro...", e)  # o e serve pra mostra onde esta o erro 

File: Aula 5/Principal.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class Principal {
    public static void main(String[] args) {
        List<Clima> lista = new ArrayList<>();
        String nomeBase = "base.csv";

        // O try-with-resources já fecha o arquivo automaticamente (igual ao 'with' do Python)
        try (BufferedReader leitor = new BufferedReader(
                new FileReader(nomeBase, StandardCharsets.UTF_8))) {
            
            String linha;
            while ((linha = leitor.readLine()) != null) {
                String[] dadosLinha = linha.split(",");
                
                Clima objClima = new Clima(dadosLinha[0],dadosLinha[1],dadosLinha[2], dadosLinha[3]);

                // Requer que o método equals() esteja implementado na classe Clima
                if (!lista.contains(objClima)) {
                    lista.add(objClima);
                }
            }

            for (Clima item : lista) {
                System.out.println(item);
            }

        } catch (Exception e) {
            System.err.println("Ocorreu algum erro... " + e.getMessage());
        }
    }
}